bin/desi_resubmit_zpix

```diff
@@ -0,0 +1,128 @@
+    """
+    Check if healpix-based redshift outputs exist
+
+    Args:
+        survey (str): main, sv1, sv2, sv3, ...
+        program (str): dark, bright, backup, other
+        healpix (int): healpix number (typically nested nside=64 matching prod)
+
+    Options:
+        allfiles (bool): if True, check all files, not just redrock
+        missing_dir_ok (bool): if True, ok if output directory is completely missing,
+            i.e. only check for files if directory exists
+
+    Returns True/False whether output files exist
+    """
+    if missing_dir_ok:
+      hpixdir = os.path.dirname(desispec.io.findfile('redrock', survey=survey,
+                                     faprogram=program, healpix=healpix))
+      if not os.path.isdir(hpixdir):
+          print(f'WARNING: missing {hpixdir}')
+          return True
+
+    if allfiles:
+        filetypes = ['spectra', 'coadd', 'redrock', 'rrdetails',
+                     'emline', 'qso_qn', 'qso_mgii']
+    else:
+        filetypes = ['redrock',]
+
+    ok = True
+    for filetype in filetypes:
+      outfile = desispec.io.findfile(filetype, survey=survey,
+                                     faprogram=program, healpix=healpix)
+      ok &= os.path.exists(outfile)
+      if not ok:
+          print(f"ERROR: missing {outfile}")
+          break
+
+    return ok
+
+#--------------------------------------------------------------------
+p = argparse.ArgumentParser(description='resubmit zpix jobs with missing outputs')
+p.add_argument('-s', '--survey', default='*',
+            help='DESI survey (main, sv1, sv3, ...)')
+p.add_argument('-p', '--program', default='*',
+            help='DESI program (dark, bright, backup, other)')
+p.add_argument('--dry-run', action='store_true',
+            help="print what to do instead of actually resubmitting jobs")
+p.add_argument('--allfiles', action='store_true',
+            help="Full check of all files including afterburners")
+p.add_argument('--missing-dir-ok', action='store_true',
+            help="Skip over completely missing output directories")
+p.add_argument('--quiet', action='store_true',
+            help="Don't report healpix for which no action is needed")
+
+args = p.parse_args()
+
+specprod_dir = desispec.io.specprod_root()
+survey = args.survey
+program = args.program
+
+#- Parse the slurm scripts to know which healpix they intended to create
+slurm_scripts = glob.glob(f'{specprod_dir}/run/scripts/healpix/{survey}/{program}/*/zpix-{survey}-{program}-*.slurm')
+
+num_resubmit = 0
+for filename in slurm_scripts:
+    basename = os.path.basename(filename)
+
+    healpix = list()
+    with open(filename) as fx:
+        for line in fx.readlines():
+            if line.startswith('srun ') and line.count(' --healpix ')>0:
+                tmp = line.split()
+                i = tmp.index('--healpix')+1
+                healpix = list()
+                while tmp[i] != '--expfiles':
```

Comment:
I really like the idea of using the zproc parser, but unfortunately the script includes `--starttime $(date +%s)` which gets evaluated by bash before passing into python, and that breaks the ability for us to parse the command line in the script directly.

However, I did update the code to be more robust when parsing `--healpix ...`: instead of looking for `--exptime` it now looks for any new option that starts with `--` or it hits the end of the list.  In the next (post-iron) update to healpix redshifts, I'd like to introduce the equivalent of the processing tables so that we don't have to do this fragile parsing in the first place.